### ISSUE TITLE
A small issue when using your code

### ISSUE THREAD

**Issue by summersnowfish** [XiYe20/VPTR#8]
Hello, I really admire your work. However, I encountered a small issue when using your code. In the following code, the line "x1 + temporal_pos_embed[:, None, :]" seems to be incorrect because the shape of x1 is ((T, NHW, C)), while the shape of "temporal_pos_embed[:, None, :]" is (T, 1, C). Moreover, the T dimension of the two tensors is different, so they cannot be directly added. Have you encountered this problem before? Is it possible that the issue is with my data? I would greatly appreciate it if you could spare some time to clarify my doubts. This code is located in the forward method of the class VidHRFormerBlockEnc(nn.Module).

def forward(self, x, local_window_pos_embed, temporal_pos_embed):
        """
        x: (N, T, H, W, C)
        local_window_pos_embed: (window_size, window_size, C)
        temporal_pos_embed: (T, C)
        Return: (N, T, H, W, C)
        """
        print(x.shape)
        print(temporal_pos_embed.shape)
        N, T, H, W, C = x.shape
        x = x + self.drop_path(self.SLMHSA(self.norm1(x), local_window_pos_embed)) #spatial local window self-attention, and skip connection
        
        #Conv feed-forward, different local window information interacts
        x = x + self.drop_path(self.SpatialFFN(self.norm2(x)))#(N, T, H, W, C)

        #temporal attention
        x = x.permute(1, 0, 2, 3, 4).reshape(T, N*H*W, C)
        x1 = self.norm3(x)
        if self.far:
            #create attention mask for temporal self-attention
            attn_mask = torch.triu(torch.ones(T, T), diagonal = 1) == 1         
            x = x + self.drop1(self.temporal_MHSA(x1 + temporal_pos_embed[:, None, :], 
                               x1 + temporal_pos_embed[:, None, :], 
                               x1, 
                               attn_mask = attn_mask.to(x1.device))[0])

**Comment by XiYe20** [comment]
Hi, for "x1 + temporal_pos_embed[:, None, :]", it is not a bug, we do this by "broadcast", you may check more information about PyTorch broadcast. I never encountered the problem that "T" of x and "T" of temporary_pos_embed are different, probably it relates to your own experiment setup?

**Comment by summersnowfish** [comment]
Thanks for your reply!
